webapp/src/pages/offer/components/OfferComponent.js

```diff
@@ -31,33 +36,71 @@ class Offer extends Component {
     };
   };
 
-  buttonSubmit() {
-    const { offer, accepted, rejected, rejected_reason } = this.state;
+  buttonSubmit(candidate_response) {
+    const { offer, rejected_reason } = this.state;
 
     offerServices
       .updateOffer(
         offer.id,
         DEFAULT_EVENT_ID,
-        accepted,
-        rejected,
-        rejected_reason
+        candidate_response,
+        candidate_response? "" : rejected_reason
```

Comment:
Consider adding a check for candidate_response being null, I know you call buttonSubmit and haven't passed a null but others might.

Reply:
candidate_response having a null isn't a problem is this case,I think... cause if its a null it just means the user hasn't responded?

Reply:
I get you, I am referring to the ```candidate_response? "" : rejected_reason``` line. If candidate_response is null, then rejected_reason will be passed, which would most prob be a null. 

Also, having a check in this function to see if candidate response is null (if it is then we just break out), will prevent sending null responses to updateOffer. It is more of a precautionary thing.